### PR TITLE
Added tests to cover more of the util functions

### PR DESCRIPTION
**Pull request by haniffm**
The tests are now covering more parts of the following functions:
- util.nested_lookup
- util.list_files

### CODE REVIEW

File: ESSArch_Core/test_util.py [hunk 1]
```diff
@@ -173,3 +177,155 @@ def test_get_schema_from_file_with_bad_input(self):
 
         with self.assertRaises(etree.XMLSyntaxError):
             getSchemas(filename=fp)
+
+
+class NestedLookupTest(TestCase):
```
Comment (OskarPersson) [inline ESSArch_Core/test_util.py]: Should also include tests for multiple occurrences of the key, and a test for when the key doesn't exist anywhere in the document

File: ESSArch_Core/test_util.py [hunk 2]
```diff
@@ -173,3 +177,155 @@ def test_get_schema_from_file_with_bad_input(self):
...
+        ]
+
+        self.assertEqual(42, next(nested_lookup('my_key', my_list)))
+
+
+class ListFilesTest(TestCase):
+
+    def setUp(self):
+        self.root = os.path.dirname(os.path.realpath(__file__))
+        self.datadir = os.path.join(self.root, "datadir")
```
Comment (OskarPersson) [inline ESSArch_Core/test_util.py]: Use `self.addCleanup` to ensure this is always deleted

File: ESSArch_Core/test_util.py [hunk 3]
```diff
@@ -173,3 +177,155 @@ def test_get_schema_from_file_with_bad_input(self):
...
+            self.assertEqual(data_type, "file")
+            self.assertEqual(data_size, 1)
+            self.assertEqual(type(data_modified), datetime.datetime)
+
+    def test_list_files_path_to_file_in_tar(self):
+        file_path = self.create_archive_file('tar')
+        new_folder = os.path.join(file_path, "./0.txt")  # TODO: bug in shutil for tar is adding an extra './'
+
+        resp = list_files(new_folder)
+        self.assertEqual(resp.status_code, 200)
```
Comment (OskarPersson) [inline ESSArch_Core/test_util.py]: This is now testing the result of `generate_file_response`. Instead, mock `generate_file_response` and ensure that it is called with the correct arguments. See the following tests for an example: https://github.com/ESSolutions/ESSArch_Core/blob/b01689ebcc2faddbd32664039563f0e77622cb62/ESSArch_Core/WorkflowEngine/tests/test_tasks.py#L37

File: ESSArch_Core/test_util.py [hunk 4]
```diff
@@ -173,3 +177,155 @@ def test_get_schema_from_file_with_bad_input(self):
...
+        with self.assertRaises(NotFound):
+            list_files(new_folder)
+
+    def test_list_files_path_to_file_in_zip(self):
+        file_path = self.create_archive_file('zip')
+        new_folder = os.path.join(file_path, "0.txt")
+
+        resp = list_files(new_folder)
+
+        self.assertEqual(resp.status_code, 200)
```
Comment (OskarPersson) [inline ESSArch_Core/test_util.py]: This is now testing the result of `generate_file_response`. Instead, mock `generate_file_response` and ensure that it is called with the correct arguments. See the following tests for an example: https://github.com/ESSolutions/ESSArch_Core/blob/b01689ebcc2faddbd32664039563f0e77622cb62/ESSArch_Core/WorkflowEngine/tests/test_tasks.py#L37

File: ESSArch_Core/test_util.py [hunk 5]
```diff
@@ -210,23 +210,53 @@ def test_nested_lookup_dict_in_list(self):
...
+            "second_layer": [
+                {"key_3": 3},
+                {"my_key": 5},
+                {"key_4": 4},
+                {"my_key": 7},
+            ]
+        }
+
+        found_values = list(nested_lookup('my_key', my_list))
+        self.assertEqual(len(found_values), 4)
```
Comment (OskarPersson) [inline ESSArch_Core/test_util.py]: Replace each assertion with a single [`assertCountEqual`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertCountEqual)

Comment (haniffm) [reply ESSArch_Core/test_util.py]: Great suggestion! :)

**Pull request merged**